components/sidebar/ObserverSections.js

```diff
@@ -0,0 +1,56 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { ObserverSection } from 'react-components';
+
+const ObserverSections = ({ list }) => {
+    const [intersectionRatios, setIntersectionRatios] = useState(
+        Array(list.length)
+            .fill(1)
+            .fill(0, 1)
+    );
+
+    useEffect(() => {
+        // find id corresponding to biggest intersection ratio
+        const indexOfMax = (arr) => arr.findIndex((elm) => elm === Math.max(...arr));
```

Comment:
Move `indexOfMax` outside of the `useEffect`. Maybe even outside this file inside a helper file.

Reply:
Done both!